File: pairs.cpp
vector<string> split_string(string);

// Complete the pairs function below.
int pairs(int k, vector<int> arr) {
    int c=0,i=0,j=1,n;
    n = arr.size();
    sort(arr.begin(),arr.end()); 
    while(i<n && j<n)
    {
        if(i!=j && arr[j]-arr[i]==k)
        {
            c+=1;
            i++;
        }
        else if(arr[j]-arr[i]<k)
        {
            j++;
        }
        else
            i++;
    }
    return c;

}

File: pairs.py
def pairs(k, arr):
    c = 0
    i = 0
    j = 1
    arr.sort()
    while i<len(arr) and j<len(arr):
        if i!=j and arr[j]-arr[i] == k:
            c+=1
            print(arr[j],arr[i])
            i+=1
        elif arr[j]-arr[i]<k:
            j+=1   
        else:
            i+=1

    return c
